src/aap_eda/services/ruleset/activation_kubernetes.py

```diff
@@ -338,32 +349,34 @@ def run_activation_job(
                     self.delete_secret(
                         secret_name=secret_name,
                         namespace=namespace,
-                        job_name=job_name,
                     )
+                    logger.info(f"Secret: {secret_name} is deleted.")
                 # remove service(s) if created
                 self.delete_services(
                     namespace=namespace,
                     job_name=job_name,
                 )
+                logger.info(f"Service: {job_name} is deleted.")
 
-    def delete_secret(self, secret_name, namespace, job_name) -> None:
-        # wait until job is done
-        while True:
-            ret = client.BatchV1Api().list_namespaced_job(
+    def delete_secret(self, secret_name, namespace) -> None:
```

Comment:
Minor concern: with this change with the labels, any secret created before the upgrade will not be removed. 
Is there a reason to do it this way instead of capturing the exception?

Reply:
Good catch, rewrote the delete_secret logic